velox/dwio/dwrf/common/PagedInputStream.cpp

```diff
@@ -188,6 +190,16 @@ void PagedInputStream::BackUp(int32_t count) {
       outputBufferPtr_ != nullptr,
       "Backup without previous Next in ",
       getName());
+  if (state_ == State::ORIGINAL) {
+    VELOX_CHECK(
```

Comment:
nit: perhaps, split this into two checks so that we have a better idea which boundary is violated in case of failure